### PR TITLE
Apache Spark --without-hadoop flag

### PR DESCRIPTION
**Pull request by iostat**
Apache Spark has a `-without-hadoop` distribution which can sometimes be handy, such as in the case when you want to use Hadoop 2.7 features, but no `apache-spark-VER-bin-hadoop2.7` distribution exists. This patch adds a --without-hadoop option to the formula to allow installing the `-without-hadoop` dist, and also provides a slightly safer version of the existing test.

### CODE REVIEW

File: Library/Formula/apache-spark.rb [hunk 1]
```diff
@@ -18,6 +25,9 @@ def install
   end
 
   test do
-    system "#{bin}/spark-shell <<<'sc.parallelize(1 to 1000).count()'"
+    # spark-shell has a REPL-history type of thing but it needs to write 
+    # to ~/.spark_history... not necessarily sandbox-friendly! But --driver-java-options
+    # lets us pass JVM flags like -D to let us override Java/Scala system properties!
+    system "#{bin}/spark-shell --driver-java-options '-Duser.home=/tmp/' <<<'sc.parallelize(1 to 1000).count()'"
```
Comment (DomT4) [inline Library/Formula/apache-spark.rb]: Just use `ENV.java_cache` here. It sets the user home. This is a common enough problem that it has an official Homebrew workaround :tada:

File: Library/Formula/apache-spark.rb [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
   desc "Engine for large-scale data processing"
   homepage "https://spark.apache.org/"
-  url "https://www.apache.org/dyn/closer.lua?path=spark/spark-1.6.0/spark-1.6.0-bin-hadoop2.6.tgz"
   version "1.6.0"
-  sha256 "439fe7793e0725492d3d36448adcd1db38f438dd1392bffd556b58bb9a3a2601"
   head "https://github.com/apache/spark.git"
-
   bottle :unneeded
 
+  option "without-hadoop", "Use the -without-hadoop distribution of Spark (assumes you have a local installation of hadoop)"
```
Comment (DomT4) [inline Library/Formula/apache-spark.rb]: I'm _pretty sure_ this is going to break our logic around the way we expect formulae to be structured, to be honest.

Comment (iostat) [inline Library/Formula/apache-spark.rb]: I had a feeling that would be the case. Perhaps it's better to split it off into a apache-spark-without-hadoop formula? I figured this scenario wouldn't affect most users (as there is invariably some manual config editing involved when running the "user-provided hadoop" version of Spark), and really only thought it was worth submitting for the people who'd like a homebrew-managed Spark install.

I could alternatively make a new formula that installs the -without-hadoop dist by default, depends on Homebrew's on `hadoop` formula, and does the configuration changes necessary to integrate it. However that *might* involve editing the Hadoop formula as well, as it doesn't seem to link its `/etc` or `/usr/local/share` files into any central version-agnostic directory. Although it's not strictly
necessary to have them linked for a functional Spark + Hadoop setup, it *does* allow you to use nice things like Hadoop extras (such as S3 and Azure Blob HDFS backends) that aren't in the default `$(hadoop classpath)`

Comment (apjanke) [inline Library/Formula/apache-spark.rb]: This sounds like enough of a specialty case that we wouldn't want to include it in core Homebrew, especially not as a separate formula. Putting this is a [user-maintained Tap](https://github.com/Homebrew/brew/blob/master/share/doc/homebrew/How-to-Create-and-Maintain-a-Tap.md) would probably be more appropriate.

**Comment by apjanke** [comment]
Closing this out since it looks like it's not compatible with the core Homebrew formula interaction structure. And legacy-homebrew is now deprecated.

**Pull request closed without merging**